backend/analytics_server/mhq/store/repos/code.py

```diff
@@ -41,7 +43,12 @@ def update_org_repos(self, org_repos: List[OrgRepo]):
         return self.get_repos_by_ids([str(repo.id) for repo in org_repos])
 
     @rollback_on_exc
-    def update_team_repos(self, team: Team, org_repos: List[OrgRepo]):
+    def update_team_repos(
+        self,
+        team: Team,
+        org_repos: List[OrgRepo],
+        idempotency_key_raw_org_repo_map: Dict[str, RawOrgRepo] = {},
+    ):
```

Comment:
either pass data model for TeamRepos here or move this logic outside the repo layer. This is doing so much work that it is not responsible for